plugins/plugins.go

```diff
@@ -518,6 +520,21 @@ func (m *Manager) setCompiler(compiler *ast.Compiler) {
 	m.compiler = compiler
 }
 
+// GetRouter returns the managers router if set
+func (m *Manager) GetRouter() *mux.Router {
+	m.mtx.Lock()
+	defer m.mtx.Unlock()
+	return m.router
+}
+
+// SetRouter sets the managers router property which is used
+// in place of the mainRouter if set
+func (m *Manager) SetRouter(r *mux.Router) {
```

Comment:
I'd like to see this function dropped from the PR--the reason is that I'm concerned about potential issues with setting the router after initialization--it's not clear to me whether that's going to work or what unintended behaviour we could run into. I think that we can satisfy the original ask (i.e., allow plugins to customize/extend the server's router) by passing the router to the Manager on construction and then exposing the getter above.